### PR TITLE
Add new exercise: spiral-matrix

### PR DESCRIPTION
**Pull request by abo64**

### CODE REVIEW

**Comment by kytrinyx** [comment]
Hey @abo64 I think @ricemery or @ErikSchierboom would be in a better position than I am to review this.

**Comment by abo64** [comment]
@kytrinyx I unintentionally clicked on your icon. :)
Have already removed you again as reviewer, sorry for bothering you.

**Comment by kytrinyx** [comment]
Ah, no worries :)

File: exercises/spiral-matrix/example.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import scala.annotation.tailrec
```
Comment (ricemery) [inline exercises/spiral-matrix/example.scala]: Not a huge deal. But it looks like the tailrec import is not used.

**Comment by abo64** [comment]
thanks for the reviews @ErikSchierboom @ricemery

**Comment by ErikSchierboom** [comment]
Thank you for creating them! I'm busy creating generators for the C# track: https://github.com/ErikSchierboom/csharp/tree/master/generators Quite a bit of effort, but I think I have most use cases covered so it should be more straightforward

**Pull request merged**